src/app/query/bitcoin/ordinals/use-inscriptions.query.ts

```diff
@@ -14,69 +14,135 @@ import { getTaprootAddress } from './utils';
 const stopSearchAfterNumberAddressesWithoutOrdinals = 20;
 const addressesSimultaneousFetchLimit = 5;
 
-// max limit value in Hiro API
-const inscriptionsLimit = 60;
+// max limit value in Hiro API - 60
+const inscriptionsLimit = 20;
 
-async function fetchInscriptions(addresses: string) {
+interface InscriptionsQueryResponse {
+  results: InscriptionResponseItem[];
+  limit: number;
+  offset: number;
+  total: number;
+}
+
+async function fetchInscriptions(addresses: string, offset = 0) {
   const res = await fetch(
-    `https://api.hiro.so/ordinals/v1/inscriptions?limit=${inscriptionsLimit}&${addresses}`
+    `https://api.hiro.so/ordinals/v1/inscriptions?limit=${inscriptionsLimit}&offset=${offset}&${addresses}`
   );
   if (!res.ok) throw new Error('Error retrieving inscription metadata');
   const data = await res.json();
-  return data.results as InscriptionResponseItem[];
+  return data as InscriptionsQueryResponse;
 }
 
 /**
  * Returns all inscriptions for the user's current taproot account
  */
-export function useGetInscriptionsQuery() {
+export function useInscriptionsInfiniteQuery() {
   const network = useCurrentNetwork();
   const keychain = useCurrentTaprootAccountKeychain();
+  const addressIndexCounter = createCounter(0);
+  const searchParams = new URLSearchParams(document.location.search);
+  const testInscriptionAddress = searchParams.get('testAddress');
+
+  let currentNumberOfAddressesWithoutOrdinals = 0;
```

Comment:
Confused by these `let`s used outside of the fns. Doesn't this work against react's render flow? Every render will make a new instance of this var set to 0?

Perhaps you should be using `useRef`?

Reply:
exactly , fixed